tectonic/dpx-pdfobj.c

```diff
@@ -1693,8 +1693,9 @@ apply_filter_TIFF2_1_2_4 (unsigned char *raster,
                 }
             }
         }
-        if (outbits > 0)
+        if (outbits > 0) {
             raster[k] = (outbuf << (8 - outbits)); k++;
```

Comment:
There are some other [questionable lines of code](https://github.com/tectonic-typesetting/tectonic/blob/master/tectonic/xetex0.c#L23185) though that'll need deeper understanding to be fix with confidence.